test/test_runner/base_test_runner.go

```diff
@@ -54,22 +62,9 @@ func (t *BaseTestRunner) GetAgentRunDuration() time.Duration {
 	return MinimumAgentRuntime
 }
 
-func (t *TestRunner) Run(s ITestSuite) {
-	testName := t.TestRunner.GetTestName()
-	log.Printf("Running %v", testName)
-	testGroupResult, err := t.runAgent()
-	if err == nil {
-		testGroupResult = t.TestRunner.Validate()
-	}
-	s.AddToSuiteResult(testGroupResult)
-	if testGroupResult.GetStatus() != status.SUCCESSFUL {
-		log.Printf("%v test group failed due to %v", testName, err)
-	}
-}
-
-func (t *TestRunner) runAgent() (status.TestGroupResult, error) {
+func (t *BaseTestRunner) RunAgent(T *TestRunner) (status.TestGroupResult, error) {
```

Comment:
nit - don't use capital letters for parameter names. 
```suggestion
func (t *BaseTestRunner) RunAgent(runner *TestRunner) (status.TestGroupResult, error) {
```

Also, I think it's confusing that the RunAgent function is defined at the interface level, but requires an implementation of the runner to be passed in.